clang-tools-extra/clangd/refactor/Rename.cpp

```diff
@@ -569,9 +576,28 @@ renameWithinFile(ParsedAST &AST, const NamedDecl &RenameDecl,
     //   }
     if (!isInsideMainFile(RenameLoc, SM))
       continue;
-    if (auto Err = FilteredChanges.add(tooling::Replacement(
-            SM, CharSourceRange::getTokenRange(RenameLoc), NewName)))
-      return std::move(Err);
+    if (std::optional<std::string> Identifier = NewName.getSinglePiece()) {
+      tooling::Replacement NewReplacement(
+          SM, CharSourceRange::getTokenRange(RenameLoc), *Identifier);
+      if (auto Err = FilteredChanges.add(NewReplacement))
+        return std::move(Err);
```

Comment:
We can’t early return because there’s the surrounding loop

```cpp
for (SourceLocation Loc : findOccurrencesWithinFile(AST, RenameDecl)) {
```

We could `continue` in the `if` block. But I don’t really like that.